ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/presenter/OntologiesPresenterImpl.java

```diff
@@ -449,4 +462,40 @@ public OntologyHierarchy getSelectedHierarchy() {
         return view.getSelectedHierarchy();
     }
 
+    @Override
+    public void onDeleteOntologyButtonClicked(final DeleteOntologyButtonClickedEvent event) {
+        serviceFacade.deleteOntology(event.getOntologyVersion(), new AsyncCallback<Void>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(Void result) {
+                announcer.schedule(new SuccessAnnouncementConfig(appearance.ontologyDeleted(event.getOntologyVersion())));
+                getOntologies(false);
+            }
+        });
+    }
+
+    @Override
+    public void onDeleteHierarchy(final DeleteHierarchyEvent event) {
+        List<OntologyHierarchy> deletedHierarchies = event.getDeletedHierarchies();
+        for (final OntologyHierarchy hierarchy : deletedHierarchies) {
```

Comment:
I think it would be best to restrict the admin to single selection and only allow them to delete 1 hierarchy at a time. This could also allow us to have a hierarchy delete button somewhere in the main view, instead of requiring a separate deletion dialog.

When a hierarchy is deleted, all classes under that hierarchy are deleted and those classes may reside under other classes in the root hierarchy.

So in one case, this loop opens the possibility for the admin to select a class plus all sub-classes in its hierarchy, then a request is sent to delete that class, followed by requests to delete the sub-classes that were already deleted in the first request.

In a less obvious scenario, the admin could select a class, then in a different branch of the root hierarchy, select a class that happens to be a sub-class of the first selection, but maybe the admin didn't notice.
For example, the admin might accidentally select class B and D for deletion in the following hierarchy, but D would already be removed by deleting B:
```
        A
       / \
      B   D
     / \    \
    C   F    E
   /
  D
 /
E
```

In fact, while reviewing this PR, I discovered a bug in the service when attempting to delete a class that doesn't exist 😖 (but also 👍 for PR reviews).